Update to MegaParsec has broken Cabal install

Hi,

# Steps to Reproduce

Attempt a clean install of Idris using Cabal 2.4.0 with the latest package list.

# Expected Behavior

Idris should install cleanly without errors.

# Observed Behavior

When attempting to build package 67 of 115 : Idris.Parser.Stack the following error occurs:

```
src/Idris/Parser/Stack.hs:70:29: error:
    Not in scope: ‘P.errorPos’
    Module ‘Text.Megaparsec’ does not export ‘errorPos’.

src/Idris/Parser/Stack.hs:76:34: error:
    Not in scope: ‘P.parseErrorPretty'’
    Perhaps you meant one of these:
      ‘P.parseErrorPretty’ (imported from Text.Megaparsec),
      ‘P.parseErrorTextPretty’ (imported from Text.Megaparsec)
    Module ‘Text.Megaparsec’ does not export ‘parseErrorPretty'’.

src/Idris/Parser/Stack.hs:107:30: error:
    Not in scope: ‘P.getPosition’
    Module ‘Text.Megaparsec’ does not export ‘getPosition’.
```

# Expected fix

The Idris build config should be pinned to the correct version of MegaParsec (and/or any other packages that aren't pinned to working versions?)

Cheers,
Ed

--- a/idris.cabal
+++ b/idris.cabal
@@ -289,7 +289,7 @@ Library
                 , fingertree >= 0.1.4.1 && < 0.2
                 , haskeline >= 0.7 && < 0.8
                 , ieee754 >= 0.7 && < 0.9
-                , megaparsec >= 6.2
+                , megaparsec >= 6.2 && < 7
                 , mtl >= 2.1 && < 2.3
                 , network < 2.8
                 , optparse-applicative >= 0.13 && < 0.15

